File: Filmes-API/Filmes.DOMAIN/Entity/All/FilmeDiretor.cs
using Filmes.DOMAIN.Interface;
using System.ComponentModel.DataAnnotations.Schema;

namespace Filmes.DOMAIN.Entity.All
{
    [Table("FilmeDiretor", Schema = "All")]
    public class FilmeDiretor : IEntity
    {
        public long Id { get; set; }
        public long FilmeId { get; set; }
        public Filme Filme { get; set; }
        public long DiretorId { get; set; }
        public Diretor Diretor { get; set; }
    }
}

File: Filmes-FRONTEND/src/core/models/all/FilmeDiretor.ts
import { Diretor,Filme } from ".";

export class FilmeDiretor implements Shared.IEntity{
    id: number = 0;
    filmeId: number = 0;
    filme!: Filme;
    diretorId:number = 0;
    diretor!:Diretor;
    
    constructor(model?: FilmeDiretor){

        if(!model)
            return;

        this.id = model.id;
        this.filmeId = model.filmeId;
        this.filme = model.filme;
        this.diretorId = model.diretorId;
        this.diretor = model.diretor;
    }
}